### ISSUE TITLE
Cannot download via Google Play

### ISSUE THREAD

**Issue by XpellCZ** [ChristopherHX/mcpelauncher-manifest#171]
I have purchased Minecraft on Google Play and I am able to log in using the launcher however when I choose a version and want to download the game it throws an error that I have not purchased the game. After reading documentation, I assume there should be a pop up asking me to confirm TOS upon clicking on download button however there is no such pop up. I have tried to clear settings and login again but to no avail. I have also tried older builds. Also picture in the middle of launcher does not load. There are only animated green loading bars displayed.

Build: amd64 x86_64 618
OS: Pop!_OS 20.04 LTS
I have two factor authentication on my Google account
I have opted for Minecraft beta program
Tried to download version 1.16.200.02 (x86_64)

Thank you in advance.

**Comment by XpellCZ** [comment]
My CPU: Intel® Core™ i5 CPU M 560

**Comment by ChristopherHX** [comment]
> I have two factor authentication on my Google account

I never tested google login of the Launcher with two factor auth. At least I heard it was a problem (download not working while using 2fa) in the past and maybe also yours. Only the insecure password login is known to work. I'm not the creator of anything google play api stuff and don't know any details. It might be possible that the launcher gets an incomplete login while 2fa is enabled.

I removed "confirm TOS", older versions will prompt for it. No idea if that removal causing trouble? Maybe try getting the legacy launcher from the wiki https://mcpelauncher.readthedocs.io/en/latest/getting_started.html#appimage and it will show you a tos approval window. To see if really still have to accept it once to use the Launcher? I "confirmed TOS" over a year ago.

> I have opted for Minecraft beta program

It isn't required to opt in to the beta program to use this Launcher, only to download beta version.

> picture in the middle of launcher does not load

Well the news are broken right now, but shoulnd't cause any problems.

**Comment by XpellCZ** [comment]
> I removed "confirm TOS", older versions will prompt for it. No idea if that removal causing trouble? Maybe try getting the legacy launcher from the wiki https://mcpelauncher.readthedocs.io/en/latest/getting_started.html#appimage and it will show you a tos approval window. To see if really still have to accept it once to use the Launcher? I "confirmed TOS" over a year ago.

I am quite sure I know what the problem is. I was able to get it running although I had to download AppImage of Minecraft PE 1.16.200.02 from the internet as I was not able to download it from Google Play.

Steps I have taken:

1.  Opened up legacy launcher you have mentioned. (copied folder with 1.16.200.02 to versions folder beforehand)
2.  Logged in with my 2FA Google account where I have purchased Minecraft PE
3. Confirmed TOS pop-up
4. At that point a new device was associated in Google Play with my account. The device is called "diy"
5. Updated to build 5xx
6. Updated to build 6xx
7. Ran successfully manually installed version 1.16.200.02

The thing is that Google Play will not authenticate you until you have the device you are playing on associated with your account and that does not happen until you "physically" open up Google Play on the device and confirm TOS pop-up. I have tried this with old Android tablet I had lying around and it is exactly the process. Your device gets associated with your account in Google Play upon confirming TOS pop-up.
